test/solc/CommandLineParser.cpp

```diff
@@ -420,18 +412,24 @@ BOOST_AUTO_TEST_CASE(invalid_options_input_modes_combinations)
 	map<string, vector<string>> invalidOptionInputModeCombinations = {
 		// TODO: This should eventually contain all options.
 		{"--error-recovery", {"--assemble", "--yul", "--strict-assembly", "--standard-json", "--link"}},
-		{"--experimental-via-ir", {"--assemble", "--yul", "--strict-assembly", "--standard-json", "--link"}}
+		{"--experimental-via-ir", {"--assemble", "--yul", "--strict-assembly", "--standard-json", "--link"}},
+		{"--metadata-literal", {"--assemble", "--yul", "--strict-assembly", "--standard-json", "--link"}},
+		{"--model-checker-show-unproved", {"--assemble", "--yul", "--strict-assembly", "--standard-json", "--link"}},
+		{"--model-checker-div-mod-no-slacks", {"--assemble", "--yul", "--strict-assembly", "--standard-json", "--link"}},
+		{"--metadata-hash=swarm", {"--assemble", "--yul", "--strict-assembly", "--standard-json", "--link"}},
 	};
 
 	for (auto const& [optionName, inputModes]: invalidOptionInputModeCombinations)
 		for (string const& inputMode: inputModes)
 		{
 			stringstream sout, serr;
-			vector<string> commandLine = {"solc", optionName, "file", inputMode};
+			size_t pos = optionName.find("=");
+			string optionNameWithoutValue = (pos != string::npos) ? optionName.substr(0, pos) : optionName;
```

Comment:
I'd add an assert against the name being empty here:
```c++
soltestAssert(!optionNameWithoutValue.empty());
```

Just to make sure that if someone by mistake uses `=` at the beginning of an option name on the list above, the code just stops instead of doing something weird (and potentially not even failing).